src/apps/concepts/components/ConceptForm.tsx

```diff
@@ -119,11 +119,26 @@ const initialValues: Concept = {
 const buildInitialValues = (
   conceptClass: string = initialValues.concept_class,
   names: ConceptName[]
-) => ({
-  ...initialValues,
-  concept_class: conceptClass,
-  names
-});
+) => {
+  // sort the names
+  names.sort((n1: ConceptName, n2: ConceptName) => {
+    if (n1.locale !== n2.locale) {
+      return n1.locale < n2.locale ? -1 : 1;
+    }
+
```

Comment:
Add this to keep the locale preferred name(s) on top of the given locale:

```suggestion

if (n1.locale_preferred !== n2.locale_preferred) {
  return n2.locale_preferred ? -1 : 1;
}

```

We may want to support sorting depending on the type `ConceptName` involved ("Fully Specified" vs "Short" vs none), but I'll leave that for someone else to define.